datafusion/core/src/dataframe/mod.rs

```diff
@@ -2984,19 +2990,25 @@ mod tests {
         let df_impl = DataFrame::new(ctx.state(), df.plan.clone());
         let func = row_number().alias("row_num");
 
+        // This first `with_column` results in a column without a `qualifier`
+        let df_impl = df_impl.with_column("s", col("c2") + col("c3"))?;
+
+        // This second `with_column` then assigns `"r"` alias to the above column and the window function
```

Comment:
```suggestion
        // This second `with_column` should only alias `func` as `"r"`
```

Update the comment from explaining the bug to explaining the correct the behavior.

The comment for this test case already links to GH issue, so maybe add a link to this bug as well.

Reply:
Added